sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/UnitTestHelper.cs

```diff
@@ -95,5 +95,57 @@ public static bool IsMsTestOrNUnitTestIgnored(this IMethodSymbol method)
                     a.AttributeClass.Is(KnownType.Xunit_FactAttribute) ||
                     a.AttributeClass.Is(KnownType.Xunit_TheoryAttribute) ||
                     a.AttributeClass.Is(KnownType.LegacyXunit_TheoryAttribute));
+
+        public static KnownType FindTestAttribute(this IMethodSymbol method)
+        {
+            return method.GetAttributes()
```

Comment:
I have the feeling it would be clearer the other way around. Something like (I am not writing from the IDE so code might not compile):

```csharp
supportedTestAttributes.FirstOrDefault(testAttribute => 
	method.GetAttributes().Any(attribute => attribute.Is(testAttribute)));
```